### PR TITLE
Keep curly braces and `from` keyword in empty `import` statements

### LINKED ISSUE #7912: Curly braces and `from` keyword stripped off of `import` statements
**Prettier 2.0.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACA7hATgawGcAdKASwFsAHPGDYAXwwDNcIKMByQ9uGACzJQA5p1Kl0GAEa44AQyKlKNXHWAZUAKgyQKFBHU2Ymrdlx76AtFTlh8c4XDFQJmGfKIYYECEuq16UgwNTABhdn1YUhM2Dm5eDAgBOAwbOwcnEAAaEAgqGDJoQmRQOVw2LAAFMoRilDkANwgyABNskBlbfD4AZTShYWQYXABXOBz+GAoAGwB1QXhCNLge2rIChvWAT2RwQmKcoUI4VUrcBwo5ZGY5aeOcgCtCAA8AIXO7Xrl9ABkhOGut3uICezx6A2mcAAiiMkgCkDc7uMQDZcMdcLspHIpHBpu0qLghDBZq0BMgABwABhyBIgx1m5youwJcHRDQBOQAjrD4Gc8nUQHJCJYoHA4C1xe1ZNyyLIzhcrgigcjjhQyICkTlCBDoTz4cMxjkYNiSS0yUgAExG85kaYDcJ6RUgKDQDkgEbHAAq2LqiOBDTGAEkoBLYD0wIT8gBBEM9GBbSEa44MBhAA)
```sh
--arrow-parens avoid
--parser babel
--no-semi
--single-quote
--trailing-comma none
```

You may wonder what would be the use case for this kind of behavior. It happened to me as I wrote a generator cli for a project creating several files based on user input data and automatically applying a prettier format when those are generated. One of the files provides the empty import statement with the comment in the middle of the curly braces to invite the developer to import what he should from a precise package.

I currently use the first version commented with `// works` in the example to work around the issue.

**Input:**
```jsx
// works
import {} from 'something'

// breaks
import { /* comment */ } from 'some-package'

// breaks too
import {
  // Comment
} from 'some other package'
```

**Output:**
```jsx
// works
import {} from 'something'

// breaks
import /* comment */ 'some-package'

// breaks too
import // Comment
'some other package'

```

**Expected behavior:**
```jsx
// works
import {} from 'something'

// breaks
import { /* comment */ } from 'some-package'

// breaks too
import {
  // Comment
} from 'some other package'

```

### PR DESCRIPTION
**Pull request by fisker**
Fixes #7912

I'll fix lint problem before merge.

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/printer-estree.js [hunk 1]
```diff
@@ -1080,12 +1081,14 @@ function printPathNoParens(path, options, print, args) {
       } else if (
         (n.importKind && n.importKind === "type") ||
         // import {} from 'x'
-        /{\s*}/.test(
+        stripComments(
```
Comment (bakkot) [inline src/language-js/printer-estree.js:1084]: Is it really worth adding a new dependency just for this?

File: changelog_unreleased/javascript/pr-8016.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { } from 'foo';
+import { /* comment */ } from 'bar';
+
+// Prettier stable
+import {} from 'foo'
+import /* comment */ 'bar'
+
+// Prettier master
+import {} from "foo";
+import {} from /* comment */ "bar";
```
Comment (thorn0) [inline changelog_unreleased/javascript/pr-8016.md:15]: This is not the expected output in https://github.com/prettier/prettier/issues/7912

File: src/language-js/printer-estree.js (hunk 1 above)
Comment (thorn0) [reply src/language-js/printer-estree.js:1084]: Agree. Prettier has its own utils for this (`getNextNonSpaceNonCommentCharacterIndexWithStartIndex`).

Comment (fisker) [reply src/language-js/printer-estree.js:1084]: Good  to  know, I'll try.

File: changelog_unreleased/javascript/pr-8016.md (hunk 2 above)
Comment (fisker) [reply changelog_unreleased/javascript/pr-8016.md:15]: Don't know how to fix this, similar one

**Prettier 2.0.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzCAnABACgDpRZYD0AVJALYUIynFYDcBRZl1sdThJ5EVNdAgEoswAL4gANCAgAHGAEtoAZ2SgAhhgwQA7gAVNCVSnUAbHeoCeq6QCMM6sAGs4MAMqzHCqAHNkMDABXOGkACxgKUwB1UIV4ZU8wODcjOIUANzjLZHBlGxBvZTgMGD0HHwp1ZFQzIukAK2UADwAhB2dXN3VqABlvOGrakJBGprdvH1M4AEVAiHhB0zqQTwwijBzbdVs4UykVjG8YKIUAExhQ5AAOAAZpWW0iqIdZHIe4dfSB6QBHOfgynJjCB1MoALRQOBwU7Q-YYOB-BTwsrqCpVJA1JbDIoUBT+ILYiZTWbzAYYobSGDbE7nS5IABMlIcClMEwAwnxKjkPgBWfaBIoAFW2xkxy3SwQAklAYbA3GBDvIAIIytwwSxTRZFMRiIA)
```sh
--parser babel
```

**Input:**
```jsx
for (
  /*comment*/ ;
  /*comment*/;
  /*comment*/
) {}
```

**Output:**
```jsx
for (;;) /*comment*/ /*comment*/ /*comment*/
{}

```

**Second Output:**
```jsx
for (;;) /*comment*/ /*comment*/ /*comment*/ {}

```

Comment (thorn0) [reply changelog_unreleased/javascript/pr-8016.md:15]: You can first add these comments as "dangling" (if `node.specifiers.length === 0 && next significant character === '}'`) and then print the dangling comments inside the curly braces.

**Comment by lydell** [comment]
Might be worth checking if this “accidentally” fixes https://github.com/prettier/prettier/issues/1359

File: src/language-js/comments.js [hunk 3]
```diff
@@ -835,14 +849,16 @@ function handleImportDeclarationComments(
   options
 ) {
   if (
-    !precedingNode &&
     enclosingNode &&
     enclosingNode.type === "ImportDeclaration" &&
-    (!enclosingNode.specifiers || enclosingNode.specifiers.length === 0)
+    (!enclosingNode.specifiers || enclosingNode.specifiers.length === 0) &&
+    options.locStart(comment) > options.locStart(enclosingNode) &&
+    options.locEnd(comment) < options.locEnd(enclosingNode)
```
Comment (thorn0) [inline src/language-js/comments.js:856]: Isn't this condition always true?

**Comment by fisker** [comment]
Close in favor of #15756

**Pull request closed without merging**